packages/core/upload/server/src/content-types/file.ts

```diff
@@ -0,0 +1,140 @@
+
+export default {
+  schema: {
+    collectionName: 'files',
+    info: {
+      singularName: 'file',
+      pluralName: 'files',
+      displayName: 'File',
+      description: '',
+    },
+    options: {},
+    pluginOptions: {
+      'content-manager': {
+        visible: false,
+      },
+      'content-type-builder': {
+        visible: false,
+      },
+    },
+    attributes: {
+      name: {
+        type: 'string',
+        configurable: false,
+        required: true,
+      },
+      alternativeText: {
+        type: 'string',
+        configurable: false,
+      },
+      caption: {
+        type: 'string',
+        configurable: false,
+      },
+      width: {
+        type: 'integer',
+        configurable: false,
+      },
+      height: {
+        type: 'integer',
+        configurable: false,
+      },
+      formats: {
+        type: 'json',
+        configurable: false,
+      },
+      hash: {
+        type: 'string',
+        configurable: false,
+        required: true,
+      },
+      ext: {
+        type: 'string',
+        configurable: false,
+      },
+      mime: {
+        type: 'string',
+        configurable: false,
+        required: true,
+      },
+      size: {
+        type: 'decimal',
+        configurable: false,
+        required: true,
+      },
+      url: {
+        type: 'string',
+        configurable: false,
+        required: true,
+      },
+      previewUrl: {
+        type: 'string',
+        configurable: false,
+      },
+      provider: {
+        type: 'string',
+        configurable: false,
+        required: true,
+      },
+      provider_metadata: {
+        type: 'json',
+        configurable: false,
+      },
+      related: {
+        type: 'relation',
+        relation: 'morphToMany',
+        configurable: false,
+      },
+      folder: {
+        type: 'relation',
+        relation: 'manyToOne',
+        target: FOLDER_MODEL_UID,
+        inversedBy: 'files',
+        private: true,
+      },
+      folderPath: {
+        type: 'string',
+        minLength: 1,
+        required: true,
+        private: true,
+        searchable: false,
+      },
+    },
+    // experimental feature:
+    indexes: [
+      {
+        name: 'upload_files_folder_path_index',
+        columns: ['folder_path'],
+        type: null,
+      },
+      {
+        name: `upload_files_created_at_index`,
+        columns: ['created_at'],
+        type: null,
+      },
+      {
+        name: `upload_files_updated_at_index`,
+        columns: ['updated_at'],
+        type: null,
+      },
+      {
+        name: `upload_files_name_index`,
+        columns: ['name'],
+        type: null,
+      },
+      {
+        name: `upload_files_size_index`,
+        columns: ['size'],
+        type: null,
+      },
+      {
+        name: `upload_files_ext_index`,
+        columns: ['ext'],
+        type: null,
+      },
+    ],
+  } satisfies Partial<Struct.CollectionTypeSchema>,
```

Comment:
random thought, does it make sense to have a strict type without the other attributes that are required in the Collection type schema but are not in schema files?

Reply:
Not sure I understand the question 😬

Reply:
It seems odd that we have to define a `Partial<Struct.CollectionTypeSchema>`  schema rather than a strict one, I believe there are some fields in that type that are not necessary to include in this file.

My question was on the line of, does it make sense to introduce an InputCollectionTypeSchema?

Reply:
@Convly we def could remove the auto generated fields from the definition

Reply:
We have a SchemaDefinition that correspond to the schema files (schema before being loaded). We could use it here too if it fits.

For now it just loosen the attributes definition, but we could pick/omit/require/partial other props

Reply:
(Should be exported in Struct too)